build/build.py

```diff
@@ -372,14 +413,147 @@ def generate_options(self):
+        return 'Visual Studio 16 2019'
+
+    @property
+    def generate_options(self):
+        return super().generate_options + [
+                f'-DCMAKE_TOOLCHAIN_FILE={get_vcpkg_toolchain_file_path(self.vcpkg_root_path)}',
+                f'-DVCPKG_TARGET_TRIPLET={self._vcpkg_triplet}'
+            ]
+
+    @property
+    def build_options(self):
+        return super().build_options + ['--config', self.config]
+
+    @property
+    def _vcpkg_triplet(self):
+        """The triplet string required by vcpkg."""
+        # There is no static version of the arm or arm64 packages.
+        # if self.arch.startswith('arm'):
+        #     return f'{self.arch}-{self.platform}'
+        # else:
+        #     return f'{self.arch}-{self.platform}-static'
+
+        # We don't use the 'static' vcpkg install yet.
+        # Looks like vcpkg uses static libs by default.
+        return f'{self.arch}-{self.platform}'
+
+class MacBuildRunner(BuildRunnerBase):
+    def __init__(self, script_args):
+        super().__init__(script_args)
+        self.vcpkg_root_path = get_env_var('BUILD_VCPKGDIR') if script_args.vcpkgdir is None else script_args.vcpkgdir
+
+    @property
+    def platform(self):
+        return 'darwin'
+
+    @property
+    def compiler(self):
+        return 'clang'
+
+    @property
+    def generator(self):
+        return 'ninja'
+
+    @property
+    def generate_options(self):
+        return super().generate_options + [
+                f'-DCMAKE_TOOLCHAIN_FILE={get_vcpkg_toolchain_file_path(self.vcpkg_root_path)}',
+                f'-DVCPKG_TARGET_TRIPLET={self._vcpkg_triplet}'
+            ]
+
+    @property
+    def build_options(self):
+        return super().build_options + ['--config', self.config]
+
+    @property
+    def _vcpkg_triplet(self):
+        """The triplet string required by vcpkg."""
+        # There is no static version of the arm or arm64 packages.
+        # if self.arch.startswith('arm'):
+        #     return f'{self.arch}-{self.platform}'
+        # else:
+        #     return f'{self.arch}-{self.platform}-static'
+
+        # We don't use the 'static' vcpkg install yet.
+        # Looks like vcpkg uses static libs by default.
+        return f'{self.arch}-{self.platform}'
+
 #endregion BuildRunner Classes
 
 #region Util Functions
 
+def create_build_runner():
+    """Creates the appropriate subclass of BuildRunnerBase.
+
+    Chooses the correct BuildRunner class for the target platform.
+    Args:
+        script_args (namespace):
+            The arguments passed to the script parsed by argparse.
+
+    Returns:
+        The appropriate subclass of Build.
+    """
+
+    if is_running_on_linux():
+        script_args = LinuxArgParser().parse()
+        return LinuxBuildRunner(script_args)
+    elif is_running_on_windows():
+        script_args = WindowsArgParser().parse()
+        return WindowsBuildRunner(script_args)
```

Comment:
>WindowsBuildRunner [](start = 15, length = 18)

I think we should fail the script if agent/plugin build is requested for windows.

Reply:
I'm failing the script if windows is requested since we don't support this yet (on github)

---
In reply to: [605010148](https://github.com/microsoft/do-client/pull/16#discussion_r605010148) [](ancestors = 605010148)